File: src/main/java/rails/game/RailsManager.java
package rails.game;

import rails.game.state.Manager;

/**
 * RailsManager adds Rails specific methods to Manager
 */
public abstract class RailsManager extends Manager implements RailsItem {

    protected RailsManager(RailsItem parent, String id) {
        super(parent, id);
    }
    
    @Override
    public RailsItem getParent() {
        return (RailsItem)super.getParent();
    }
    
    @Override
    public RailsRoot getRoot() {
        return (RailsRoot)super.getRoot();
    }
    
}

File: dotnet/RailsLib.Net/Net/Game/RailsManager.cs
using GameLib.Net.Game.State;
using System;

/**
 * RailsManager adds Rails specific methods to Manager
 */
namespace GameLib.Net.Game
{
    abstract public class RailsManager : Manager, IRailsItem
    {
        protected RailsManager(IRailsItem parent, String id) : base(parent, id)
        {
        }

        new public IRailsItem Parent
        {
            get
            {
                return (IRailsItem)base.Parent;
            }
        }


        new public RailsRoot GetRoot
        {
            get
            {
                return (RailsRoot)base.GetRoot;
            }
        }
    }
}
